### PR TITLE
chore: implement expanding volume in a faked way

### PR DESCRIPTION
**Pull request by cnut**

### CODE REVIEW

File: pkg/csi/controllerserver.go [hunk 1]
```diff
@@ -661,26 +661,53 @@ func (cs *controllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
...
 // ControllerExpandVolume expand volume
 func (cs *controllerServer) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
 	log.V(4).Infof("ControllerExpandVolume: called with args %+v", *req)
 
-	// Step 1: get vgName
 	volumeID := req.GetVolumeId()
+	volSizeBytes := int64(req.GetCapacityRange().GetRequiredBytes())
+
+	// Step 1: get vgName
+	//volumeID := req.GetVolumeId()
```
Comment (zjx20) [inline pkg/csi/controllerserver.go:686]: deadcode

File: pkg/csi/controllerserver.go [hunk 2]
```diff
@@ -661,26 +661,53 @@ func (cs *controllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
...
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "ControllerExpandVolume: fail to get node name of pv %s: %s", pv.Name, err.Error())
 	}
+
+	volumeType, err := getVolumeType(pv)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "ControllerExpandVolume: fail to get volumeType: %s", err.Error())
+	}
+	if volumeType == string(pkg.VolumeTypeHostPath) {
+		log.Infof("assume we are hostpath pv, fake to expand %s successfully", volumeID)
```
Comment (zjx20) [inline pkg/csi/controllerserver.go:700]: leave a TODO

File: pkg/csi/controllerserver.go [hunk 3]
```diff
@@ -661,26 +661,53 @@ func (cs *controllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
...
+	}
+
 	vgName := utils.GetVGNameFromCsiPV(pv)
 	if vgName == "" {
 		return nil, status.Errorf(codes.Internal, "ControllerExpandVolume: fail to get vgName of pv %s", pv.Name)
 	}
 
 	// Step 2: check whether the volume can be expanded
-	volSizeBytes := int64(req.GetCapacityRange().GetRequiredBytes())
+	// volSizeBytes := int64(req.GetCapacityRange().GetRequiredBytes())
```
Comment (zjx20) [inline pkg/csi/controllerserver.go:710]: deadcode

File: pkg/csi/nodeserver.go [hunk 4]
```diff
@@ -439,6 +439,9 @@ func (ns *nodeServer) resizeVolume(ctx context.Context, volumeID, targetPath str
 	}
 
 	switch volumeType {
+	case string(pkg.VolumeTypeHostPath):
+		log.Infof("NodeExpandVolume:: fake to expand hostpath successful volumeId: %s", volumeID)
```
Comment (zjx20) [inline pkg/csi/nodeserver.go:443]: leave a TODO

**Review by zjx20 — APPROVED** [review]

File: pkg/csi/controllerserver.go [hunk 5]
```diff
@@ -661,26 +661,53 @@ func (cs *controllerServer) DeleteSnapshot(ctx context.Context, req *csi.DeleteS
 	return &csi.DeleteSnapshotResponse{}, nil
 }
 
+func getVolumeType(pv *v1.PersistentVolume) (volumeType string, err error) {
+	// Get volumeType
+	volumeType = string(pkg.VolumeTypeUnknown)
+	if pv != nil && pv.Spec.CSI != nil {
+		if value, ok := pv.Spec.CSI.VolumeAttributes["volumeType"]; ok {
```
Comment (gufeijun) [inline pkg/csi/controllerserver.go:668]: Use defined global variable 'VolumeTypeTag' instead of hard code "volumeType"

**Review by gufeijun — APPROVED** [review]

File: pkg/csi/controllerserver.go (hunk 5 above)
Comment (cnut) [reply pkg/csi/controllerserver.go]: done

File: pkg/csi/controllerserver.go (hunk 2 above)
Comment (cnut) [reply pkg/csi/controllerserver.go:700]: done

File: pkg/csi/nodeserver.go (hunk 4 above)
Comment (cnut) [reply pkg/csi/nodeserver.go:443]: done

**Pull request merged**